### PR TITLE
feat: add mcm configuration for mod

### PR DESCRIPTION
**Pull request by CollinHerber**
* Adding MCM Configuration
* Updated target framework for additional language abilities
* Updated readme to include nm link

Can contact me on discord for additional information "Collin#9999" if you'd like.

### CODE REVIEW

**Comment by hamzah-hayat** [comment]
Also, on the note of the additions to the .sln file and .csproj, maybe we should add a gitignore for them?
It looks like theres a bunch of differences between our versions of them, bit unsure if I want to merge them.
The author change is fine though 😄

**Comment by CollinHerber** [comment]
> Also, on the note of the additions to the .sln file and .csproj, maybe we should add a gitignore for them?
> It looks like theres a bunch of differences between our versions of them, bit unsure if that will merge nicely.
> The author change is fine though 😄

Heya! Those files would need to remain unignored for framework/build changes. The only thing that should change on our individual computers would be the GameFolder section:
https://github.com/hamzah-hayat/AdditionalQuests/pull/1/files#diff-27e32143275dcc22f6e153b22bc152de40cb17fdf4db84b923077082a74f2a41R9

On another note, regarding the sln, I did see you are using Visual Studio 15. I would update that to 16 and then there would be no changes there. 

The only reason these changed drastically is because of the target framework change and the cleanup of the references (using wildcards instead of specific imports)

**Comment by CollinHerber** [comment]
> Hi Collin, thanks for the work, You were super quick!
> Would this require Harmony or MCM as a hard dependency?
> 
> I like the idea of enabling/disabling individual quests or maybe even editing values later on but I also want to keep the mod as light as possible, ideally without requiring harmony.
> Having MCM as an optional module would be good though.
> 
> I dont think MCM needs harmony, could be wrong though

This would make MCM and Harmony a hard requirement.

We could do some work to not require MCM or Harmony, however I'm not sure if that's worth the additional effort.
The reason I say this is everybody using mods in this community have these added. 

Let me know which route you'd like to go.

**Comment by hamzah-hayat** [comment]
Alright, the visual studio project file changes makes sense to me, thanks
With regards to usage of harmony and MCM, I would prefer we descope harmony completely (until we need it).
Given the recent changes to mod support code wise by Talesworlds I feel like long-term it will be better to keep things as simple as possible, things that Harmony was used for may no longer be relevant.

For MCM, if we can keep it optional that would also be good. Eg having a config file that is loaded into memory that is also mapped to MCM options. This way you could change the config manually (via text editor) or optionally having MCM installed to change them in game.

It might be extra work now, but I think keeping everything as decoupled as possible is best.
Especially since theres not many options right now that we need (just an off/on switch for each quest).

Thanks for the hard work though, I appreciate the time you've taken to put this all together

**Comment by CollinHerber** [comment]
Sure, I agree with your logic. I will go ahead and decouple the hard requirements.

**Comment by CollinHerber** [comment]
@hamzah-hayat https://github.com/hamzah-hayat/AdditionalQuests/pull/1/commits/c4e0b3d7966fb69f228431911a09bd51bd3eaae8
Hard dependencies have been removed.

File: AdditionalQuestsCode/AdditionalQuestsCode/Utils/Logging.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+using TaleWorlds.Library;
+using TaleWorlds.Localization;
+
+namespace AdditionalQuestsCode.Utils
+{
+    public static class Logging
+    {
+        public static bool logToFile = true;
+        public static bool Debug = true;
+        public static string PrePrend = "ScottiesQuests";
```
Comment (hamzah-hayat) [inline AdditionalQuestsCode/AdditionalQuestsCode/Utils/Logging.cs:15]: Rename to "AdditionalQuests"

Comment (CollinHerber) [reply AdditionalQuestsCode/AdditionalQuestsCode/Utils/Logging.cs]: Resolved

File: AdditionalQuestsCode/AdditionalQuestsCode/AdditionalQuestsCode.csproj [hunk 2]
```diff
@@ -1,88 +1,70 @@
...
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+    <PackageReference Include="Bannerlord.ButterLib" Version="1.0.21" IncludeAssets="compile" />
+    <PackageReference Include="Bannerlord.UIExtenderEx" Version="2.1.1" IncludeAssets="compile" />
+    <PackageReference Include="Bannerlord.MCM" Version="4.3.4" IncludeAssets="compile" />
+    <PackageReference Include="Newtonsoft.Json" Version="11.0.2">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>compile</IncludeAssets>
+    </PackageReference>
```
Comment (hamzah-hayat) [inline AdditionalQuestsCode/AdditionalQuestsCode/AdditionalQuestsCode.csproj:68]: What downloads do we need to compile?
the URL links would be helpful, pop it in the readme under a "Dependencies" header

**Comment by hamzah-hayat** [comment]
Rest looks good to me, happy to merge

File: AdditionalQuestsCode/AdditionalQuestsCode/AdditionalQuestsCode.csproj (hunk 2 above)
Comment (CollinHerber) [reply AdditionalQuestsCode/AdditionalQuestsCode/AdditionalQuestsCode.csproj:68]: If you go look at what is bundled already with the vanilla game, Newtonsoft.Json is one of them. We do not need to handle bundling this as it's already provided.

Comment (hamzah-hayat) [reply AdditionalQuestsCode/AdditionalQuestsCode/AdditionalQuestsCode.csproj:68]: ah, I meant the other libraries
These ones:
Bannerlord.MCM - https://www.nexusmods.com/mountandblade2bannerlord/mods/612
Bannerlord.UIExtenderEx - https://www.nexusmods.com/mountandblade2bannerlord/mods/2102
Bannerlord.ButterLib - https://www.nexusmods.com/mountandblade2bannerlord/mods/2018
Bannerlord.BUTR.Shared - https://www.nexusmods.com/mountandblade2bannerlord/mods/2513
Just double checking that we have the same ones for these

Comment (hamzah-hayat) [reply AdditionalQuestsCode/AdditionalQuestsCode/AdditionalQuestsCode.csproj:68]: Ah, nvm I see that nuget handles all those packages, nice

**Pull request merged**